object-cache.php

```diff
@@ -1000,46 +1048,80 @@ protected function _connect_redis() {
 			}
 		}
 
-		$this->redis = new Redis;
-
 		if ( file_exists( $redis_server['host'] ) && 'socket' === filetype( $redis_server['host'] ) ) { //unix socket connection
 			//port must be null or socket won't connect
 			$port = null;
 		} else { //tcp connection
 			$port = ! empty( $redis_server['port'] ) ? $redis_server['port'] : 6379;
 		}
-		$this->redis->connect( $redis_server['host'], $port, 1, null, 100 ); // 1s timeout, 100ms delay between reconnections
-		$keys_methods = array(
-			'auth'     => 'auth',
-			'database' => 'select',
+
+		$defaults = array(
+			'host' => $redis_server['host'],
+			'port' => $port,
+			'timeout' => 1000, // I multiplied this by 1000 so we'd have a common measure of ms instead of s and ms, need to make sure this gets divided by 1000
+			'retry_interval' => 100,
+		);
+		// 1s timeout, 100ms delay between reconnections
+
+		// merging the defaults with the original $redis_server enables any
+		// custom parameters to get sent downstream to the redis client.
+		return array_replace_recursive( $redis_server, $defaults );
+	}
+
+	public function build_client_connection( $client_parameters ) {
```

Comment:
Generally, I prefer not to expose public methods unless we absolutely need to. Public methods become APIs we promise to support indefinitely.